### DISCUSSION TITLE
yarn workspace foreach exclude not working with multiple package names

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#2629] [Q&A] [answered]
I am trying to use the `yarn workspace foreach` command with the `--exclude` flag. The exact command I am running is this:

`yarn workspaces foreach -p --exclude "['workspace-one', 'workspace-two']" run build`

The issue is the command I am passing in is being run on the second argument in the match array. So in other words, `run build` is being run on `workspace-two`. Can someone please let me know if my syntax is incorrect here? I'm not sure if this is a bug or on me. Without examples, it's difficult to know the correct syntax.

**Comment by merceyz** [comment, marked as answer]
The syntax for arrays in the CLI looks like this:
```sh
yarn workspaces foreach -p --exclude workspace-one --exclude workspace-two run build
```

**Reply by merceyz** [reply]
Also, if you want to build `workspace-a` and all workspaces it depends on you can use the `-R` flag https://yarnpkg.com/cli/workspaces/foreach#options-R%2C-recursive

```
yarn workspace workspace-a workspaces foreach -ptvR run build
```

**Reply by wdfinch** [reply]
Gotcha. Thanks for the answer. I was especially confused because the docs link to this [library](https://github.com/micromatch/micromatch) when providing details on how to use the command. This library lists some examples like: 

`console.log(micromatch(['foo', 'bar', 'baz', 'qux'], ['f*', 'b*'])) //=> ['foo', 'bar', 'baz']`

which lead me down the wrong path. It would be really helpful if an example of `--exclude` and `--include` could be added to the docs.

**Reply by wdfinch** [reply]
That's really helpful. Thanks. Again would love to see some more examples in the docs.

**Reply by merceyz** [reply]
Yeah, that's the docs for what is a valid glob pattern, the docs for "how arrays work in the CLI" would be here https://mael.dev/clipanion/docs/api/option#optionarray but yeah, should probably document it better. A PR improving the docs is always welcome :)

Relevant sections to update:
https://github.com/yarnpkg/berry/blob/76a47b3a800e54750836ecded3610928918b214a/packages/plugin-workspace-tools/sources/commands/foreach.ts#L134-L136
https://github.com/yarnpkg/berry/blob/76a47b3a800e54750836ecded3610928918b214a/packages/plugin-workspace-tools/sources/commands/foreach.ts#L85-L94

**Reply by wdfinch** [reply]
Gotcha. A link to that article in the docs would be helpful. Thanks for the help here. I would make a PR, but I didn't know how it worked lol. Maybe once I'm more familiar with this, I can improve it.

**Reply by hirenumradia** [reply]
+1 for examples in the doc

**Comment by Robula** [comment]
So what is the correct syntax? I've tried:
`yarn workspaces foreach -pv --exclude 'root,*script*,@types/*,*eslint*'`
`yarn workspaces foreach -pv --exclude '{root,*script*,@types/*,*eslint*}'`
`yarn workspaces foreach -pv --exclude '[root,*script*,@types/*,*eslint*]'`

**Reply by merceyz** [reply]
Neither of them, see the accepted answer

**Comment by sonam-diwate-invimatic** [comment]
yarn workspaces foreach -p --exclude workspace-one --exclude workspace-two run build

this is not working now
